iteratee/src/main/scala/io/finch/iteratee/package.scala

```diff
@@ -0,0 +1,91 @@
+package io.finch
+
+import cats.Eval
+import com.twitter.finagle.http.Response
+import com.twitter.io._
+import com.twitter.util.Future
+import io.catbird.util._
+import io.finch.internal._
+import io.finch.items.RequestItem
+import io.iteratee.{Enumerator, Iteratee}
+import io.iteratee.twitter.FutureModule
+import shapeless.Witness
+
+/**
+  * Iteratee module
+  */
+package object iteratee extends IterateeInstances {
+
+  import syntax._
+
+  private[finch] def enumeratorFromReader(reader: Reader): Enumerator[Future, Buf] = {
+    reader.read(Int.MaxValue).intoEnumerator.flatMap {
+      case None => empty[Buf]
+      case Some(buf) => enumOne(buf).append(enumeratorFromReader(reader))
+    }
+  }
+
+  /**
+    * An evaluating [[Endpoint]] that reads a required chunked streaming binary body, interpreted as
+    * an `Enumerator[Future, A]`. The returned [[Endpoint]] only matches chunked (streamed) requests.
+    */
+  def enumeratorBody[A, CT <: String](implicit decode: Enumerate.Aux[A, CT]): Endpoint[Enumerator[Future, A]] = {
+    new Endpoint[Enumerator[Future, A]] {
+      final def apply(input: Input): Endpoint.Result[Enumerator[Future, A]] = {
+        if (!input.request.isChunked) {
+          EndpointResult.Skipped
+        } else {
+          val req = input.request
+          def enum = enumeratorFromReader(req.reader).ensureEval(Eval.later(Future.value(req.reader.discard())))

```

Comment:
I'd probably prefer if `enumeratorFromReader` returned an enumerator that already ensures to discard the reader.